alacritty/src/window_context.rs

```diff
@@ -184,8 +191,8 @@ impl WindowContext {
         }
 
         // Live title reload.
-        if !config.window.dynamic_title || old_config.window.title != config.window.title {
-            self.display.window.set_title(&config.window.title);
+        if !config.window.dynamic_title || self.identity.title != config.window.identity.title {
+            self.display.window.set_title(&config.window.identity.title);
         }
```

Comment:
Is it me or should this be `config.window.dynamic_title && ...`?

Also since `dynamic_title` is only ever activated when the options does **not** have a title passed, we should probably be able to use the old logic and not store the `identity` on the `WindowContext` unless we remove the override code from `override_config`?